executor/aggfuncs/func_max_min.go

```diff
@@ -14,122 +14,133 @@
 package aggfuncs
 
 import (
-	"container/heap"
 	"unsafe"
 
+	"github.com/pingcap/errors"
 	"github.com/pingcap/tidb/sessionctx"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/types/json"
 	"github.com/pingcap/tidb/util/chunk"
 	"github.com/pingcap/tidb/util/stringutil"
 )
 
-type maxMinHeap struct {
-	data    []interface{}
-	h       heap.Interface
-	varSet  map[interface{}]int64
-	isMax   bool
-	cmpFunc func(i, j interface{}) int
+// newDeque inits a new Deque
+func newDeque(isMax bool, cmpFunc func(i, j interface{}) int) *Deque {
+	return &Deque{[]Pair{}, isMax, cmpFunc}
 }
 
-func newMaxMinHeap(isMax bool, cmpFunc func(i, j interface{}) int) *maxMinHeap {
-	h := &maxMinHeap{
-		data:    make([]interface{}, 0),
-		varSet:  make(map[interface{}]int64),
-		isMax:   isMax,
-		cmpFunc: cmpFunc,
-	}
-	return h
+// Deque is an array based double end queue
+type Deque struct {
```

Comment:
```
type deque struct {
	items   []Pair
	isMax   bool
	cmpFunc func(i, j interface{}) int
}
```

Because the `Deque` is only used in the `aggfuncs` package.